libs/barista-components/sunburst-chart/src/sunburst-chart.ts

```diff
@@ -194,110 +235,47 @@ export class DtSunburstChart {
-    const origin = this._getOriginFromSlice(node);
-
-    this._dismissOverlay();
-    if (origin) {
-      //  TODO: lukas.holzer
-      // Bazel cannot resolve the extends of the interface properly and
-      // throws error: Property 'data' does not exist on type 'DtSunburstChartSlice'.
-      // @ts-ignore
-      this._createOverlay(origin, node.data);
-    }
-  }
-
-  /** Creates the overlay and attaches it. */
-  private _createOverlay(
-    origin: { x: number; y: number },
-    node: DtSunburstChartTooltipData,
+  /**
+   * @internal
+   * Handles the mouseEnter on a slice.
+   * Creates an overlay if it is necessary.
+   */
+  _handleOnMouseEnter(
+    event: MouseEvent,
+    slice: DtSunburstChartTooltipData,
   ): void {
-    // If we do not have an overlay defined, we do not need to attach it
-    if (!this._overlay) {
-      return;
-    }
-
-    // Create the template portal
-    if (!this._portal) {
-      // tslint:disable-next-line: no-any
-      this._portal = new TemplatePortal<any>(
-        this._overlay,
-        this._viewContainerRef,
-        { $implicit: node },
-      );
-    }
-
-    const positionStrategy = this._overlayService
-      .position()
-      .flexibleConnectedTo(origin)
-      .setOrigin(origin)
-      .withPositions([
-        {
-          originX: 'center',
-          originY: 'center',
-          overlayX: 'center',
-          overlayY: 'center',
-        },
-      ])
-      .withFlexibleDimensions(true)
-      .withPush(false)
-      .withGrowAfterOpen(true)
-      .withViewportMargin(0)
-      .withLockedPosition(false);
-
-    const overlayConfig = new OverlayConfig({
-      positionStrategy,
-      panelClass: OVERLAY_PANEL_CLASS,
-      backdropClass: 'cdk-overlay-transparent-backdrop',
-    });
-    this._overlayRef = this._overlayService.create(overlayConfig);
-
-    // If the portal is not yet attached to the overlay, attach it.
-    if (!this._overlayRef.hasAttached()) {
-      this._overlayRef.attach(this._portal);
+    if (this._overlay && !this._overlayRef) {
+      this._overlayRef = this._overlayService.create<
+        DtSunburstChartTooltipData
+      >(event.target as HTMLElement, this._overlay);
+      this._overlayRef.updateImplicitContext(slice);
+      this._overlayRef.updatePosition(event.offsetX, event.offsetY);
     }
-    dtSetUiTestAttribute(
-      this._overlayRef.overlayElement,
-      this._overlayRef.overlayElement.id,
-      this._elementRef,
-      this._config,
-    );
   }
-
-  /** Dismisses the overlay. */
-  private _dismissOverlay(): void {
+  /**
+   * @internal
+   * Handles the mouseMove on a slice.
+   * Updates the position of the overlay to create a mouseFollow position
+   */
+  _handleOnMouseMove(event: MouseEvent): void {
     if (this._overlayRef) {
-      this._overlayRef.detach();
-      this._portal = null;
+      this._overlayService._positionStrategy.setOrigin({
+        x: event.clientX,
+        y: event.clientY,
+      });
```

Comment:
Please check this in firefox if you haven't already. We had the problems in the past with clientX or event offsets in general not being the same on svg elements in firefox as in blink-based browsers.

Reply:
works as a charm :D